lib/datadog/core/configuration/base.rb

```diff
@@ -79,6 +79,19 @@ def to_h
             options_hash
           end
 
+          # Converts the configuration variables into a flat hash with period delimited keys
+          def to_hash(hash: to_h)
+            hash.each_with_object({}) do |(k, v), h|
+              if v.is_a?(Configuration::Options) && !(inner_hash = v.to_h).empty?
+                to_hash(hash: inner_hash).map do |h_k, h_v|
+                  h["#{k}.#{h_k}"] = h_v
+                end
+              else
+                h[k.to_s] = v
+              end
+            end
+          end
+
           def reset!
```

Comment:
I don't think a `#to_hash` that converts the configuration variables into a flat hash with period delimited keys is of interest to the `Configuration::Base`.

This seems very specific to telemetry.

I suggest this method be moved to the telemetry `Collector`.